### PR TITLE
Xunit conversion

### LINKED ISSUE #15: Migrate to xUnit
Rewrite existings tests with xUnit instead because NUnit no longer maintained.

### LINKED ISSUE #36: Some tests with mocks fail
| Test class name             | Test method name                   |
|-----------------------------|------------------------------------|
| ModbusAsciiTransportFixture | ReadRequestResponseNotEnoughBytes  |
| ModbusTransportFixture      | UnicastMessage_ErrorSlaveException |

### PR DESCRIPTION
**Pull request by richardlawley**
Migrated unit tests to Xunit.  Fixes #15 

* Required upgrading the test project to Framework 4.5 (minimum supported by Xunit).
* Required removing some mock setups which were previously failing, but being hidden by the ExpectedException attribute which is not used in Xunit (fixes #36).

### CODE REVIEW

**Comment by Maxwe11** [comment]
@richardlawley good work, thanks.
I've checked your branch locally and catch that some tests periodically fail. I think this happening because xunit 2.0 by default runs tests in parallel and Rhino mocks turned up not thread safe :smile: 
So until we couldn't resolve this we should disable parallelism. You can do this by adding the following line to *AssemblyInfo.cs* file.

    [assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly, DisableTestParallelization = true, MaxParallelThreads = 1)]

**Comment by Maxwe11** [comment]
About bb743c9. Could you revert it and tests which fail mark as

    [Fact(Skip="https://github.com/NModbus4/NModbus4/issues/36")]

**Comment by richardlawley** [comment]
I don't believe #36 is actually a problem - it's the `mocks.VerifyAll()` which failed - from what I can see the thing being tested is still fine.  I suspect the setup was copied from another test, but the use of ExpectedException prevented mocks.VerifyAll from ever being called so it was never noticed.

Let me know what you think before I do any more tidying up.

**Comment by Maxwe11** [comment]
>I don't believe #36 is actually a problem - it's the mocks.VerifyAll() which failed - from what I can see the thing being tested is still fine.

:ok: will mark your PR as fix for #36

File: NModbus4.UnitTests/NModbus4.UnitTests.csproj [hunk 1]
```diff
@@ -34,6 +35,8 @@
     <TargetFrameworkProfile />
     <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
```
Comment (Maxwe11) [inline NModbus4.UnitTests/NModbus4.UnitTests.csproj]: @richardlawley I've changed appveyor setting and now build succeeds.
Also remove these lines.

File: NModbus4.UnitTests/NModbus4.UnitTests.csproj [hunk 2]
```diff
@@ -44,6 +47,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <UseVSHostingProcess>true</UseVSHostingProcess>
+    <Prefer32Bit>false</Prefer32Bit>
```
Comment (Maxwe11) [inline NModbus4.UnitTests/NModbus4.UnitTests.csproj]: Why?

File: NModbus4.UnitTests/IO/ModbusRtuTransportFixture.cs [hunk 3]
```diff
@@ -164,7 +160,7 @@ public void ReadResponseSlaveException()
 
             mocks.ReplayAll();
 
-            Assert.IsTrue(transport.ReadResponse<ReadCoilsInputsResponse>() is SlaveExceptionResponse);
+            Assert.True(transport.ReadResponse<ReadCoilsInputsResponse>() is SlaveExceptionResponse);
```
Comment (Maxwe11) [inline NModbus4.UnitTests/IO/ModbusRtuTransportFixture.cs]: :question: `Assert.IsType`

File: NModbus4.UnitTests/NModbus4.UnitTests.csproj (hunk 2 above)
Comment (richardlawley) [inline NModbus4.UnitTests/NModbus4.UnitTests.csproj]: Unintentional - I have this set in Visual Studio as some of my work requires a 64 bit runner.  Didn't realise it updated projects I open!  The NuGetPackageImportStamp was not explicitly added either, but I suspect is a result of using VS2015 and the newest nuget, and anyone who changes a project will probably reintroduce the same thing.

File: NModbus4.UnitTests/IO/ModbusRtuTransportFixture.cs (hunk 3 above)
Comment (richardlawley) [inline NModbus4.UnitTests/IO/ModbusRtuTransportFixture.cs]: Not my test, I just changed the assert library syntax to xunit

**Comment by Maxwe11** [comment]
LGTM. Thanks for taking a look at this. I will merge the PR once the CI completes.

**Pull request merged**